miio/integrations/vacuum/roborock/vacuumcontainers.py

```diff
@@ -353,6 +484,77 @@ def complete(self) -> bool:
         return self.data["complete"] == 1
 
 
+class FloorCleanDetails(DeviceStatus):
+    """Contains details about a last cleaning run per floor."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+    @property
+    @sensor(
+        "Floor 0 clean start",
```

Comment:
Instead of hardcoding the floor numbers, maybe just override the `sensors()` to return a generated list of sensor descriptors? This way it'll be easy to extend in case some models support different number of floors.

Alternatively, if other information besides the start date is interesting, the main container could embed "FloorDetails" container?

Reply:
Alright, I made the `FloorCleanDetails` status container dynamic such that it will also work for any number of floors.
It does make it a bit hard to read, and unfortunatley Properties can not be set of an instance of a class, so I needed to also overwrite the __repr__.

The main status() already embeds the `FloorCleanDetails`.

Reply:
I also embeded CleanDetails for each floor in the FloorCleanDetails container, so other details can also be accesed.